src/server/game/Scripting/ScriptMgr.cpp

```diff
@@ -2179,220 +2188,1098 @@ void ScriptMgr::OnBeforeMailDraftSendMailTo(MailDraft* mailDraft, MailReceiver c
     FOREACH_SCRIPT(MailScript)->OnBeforeMailDraftSendMailTo(mailDraft, receiver, sender, checked, deliver_delay, custom_expiration, deleteMailItemsFromDB, sendMail);
 }
 
-AllMapScript::AllMapScript(const char* name)
-    : ScriptObject(name)
+void ScriptMgr::OnBeforeUpdatingPersonalRating(int32& mod, uint32 type)
 {
-    ScriptRegistry<AllMapScript>::AddScript(this);
+    FOREACH_SCRIPT(FormulaScript)->OnBeforeUpdatingPersonalRating(mod, type);
 }
 
-AllCreatureScript::AllCreatureScript(const char* name)
-    : ScriptObject(name)
+bool ScriptMgr::OnBeforePlayerQuestComplete(Player* player, uint32 quest_id)
 {
-    ScriptRegistry<AllCreatureScript>::AddScript(this);
+    bool ret=true;
+    FOR_SCRIPTS_RET(PlayerScript, itr, end, ret) // return true by default if not scripts
+    if (!itr->second->OnBeforeQuestComplete(player, quest_id))
+        ret=false; // we change ret value only when scripts return false
+        
+    return ret;
 }
 
-UnitScript::UnitScript(const char* name, bool addToScripts)
-    : ScriptObject(name)
+void ScriptMgr::OnBeforeStoreOrEquipNewItem(Player* player, uint32 vendorslot, uint32& item, uint8 count, uint8 bag, uint8 slot, ItemTemplate const* pProto, Creature* pVendor, VendorItem const* crItem, bool bStore)
 {
-    if (addToScripts)
-        ScriptRegistry<UnitScript>::AddScript(this);
+    FOREACH_SCRIPT(PlayerScript)->OnBeforeStoreOrEquipNewItem(player, vendorslot, item, count, bag, slot, pProto, pVendor, crItem, bStore);
 }
 
-MovementHandlerScript::MovementHandlerScript(const char* name)
-    : ScriptObject(name)
+
+bool ScriptMgr::CanJoinInArenaQueue(Player* player, uint64 BattlemasterGuid, uint8 arenaslot, BattlegroundTypeId BGTypeID, uint8 joinAsGroup, uint8 IsRated, GroupJoinBattlegroundResult& err)
 {
-    ScriptRegistry<MovementHandlerScript>::AddScript(this);
+    bool ret = true;
+
+    FOR_SCRIPTS_RET(PlayerScript, itr, end, ret) // return true by default if not scripts
+        if (!itr->second->CanJoinInArenaQueue(player, BattlemasterGuid, arenaslot, BGTypeID, joinAsGroup, IsRated, err))
+            ret = false; // we change ret value only when scripts return false
+
+    return ret;
 }
 
-SpellScriptLoader::SpellScriptLoader(const char* name)
-    : ScriptObject(name)
+bool ScriptMgr::CanBattleFieldPort(Player* player, uint8 arenaType, BattlegroundTypeId BGTypeID, uint8 action)
 {
-    ScriptRegistry<SpellScriptLoader>::AddScript(this);
+    bool ret = true;
+
+    FOR_SCRIPTS_RET(PlayerScript, itr, end, ret) // return true by default if not scripts
+        if (!itr->second->CanBattleFieldPort(player, arenaType, BGTypeID, action))
+            ret = false; // we change ret value only when scripts return false
```

Comment:
are we returnig `false` if the LAST value of `FOR_SCRIPTS_RET` is false or if ANY of them returns false?